File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
	
	static public int n, m, v;
	public static boolean[][] graph;
	public static boolean[] visitDFS, visitBFS;

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		
		n = Integer.parseInt(st.nextToken());
		m = Integer.parseInt(st.nextToken());
		v = Integer.parseInt(st.nextToken());
		
		graph = new boolean[n+1][n+1];
		
		visitDFS = new boolean[n+1];
		visitBFS = new boolean[n+1];
		
		int a, b;
		for(int i = 0; i < m; i++) {
			st = new StringTokenizer(br.readLine());
			a = Integer.parseInt(st.nextToken());
			b = Integer.parseInt(st.nextToken());
			
			graph[a][b] = true;
			graph[b][a] = true;
			
		}
		
		dfs(v);
		System.out.println();
		bfs(v);
		
		

	}
	
	public static void dfs(int i) {
		visitDFS[i] = true;
		System.out.print(i + " ");
		for(int j=1; j<n+1; j++) {
			if(graph[i][j] == true && visitDFS[j] == false) {
				dfs(j);
			}
		}
	}
	
	public static void bfs(int i) {
		Queue<Integer> q = new LinkedList<>();
		q.offer(i);
		visitBFS[i] = true;
		
		while(!q.isEmpty()) {
			int temp = q.poll();
			System.out.print(temp + " ");
			for(int j=1; j<n+1; j++) {
				if(graph[temp][j] == true && visitBFS[j] == false) {
					q.offer(j);
					visitBFS[j] = true;
				}
			}
		}
	}

}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque
# BFS 메서드 정의
def bfs(graph, start, visited):
    queue = deque([start])
    visited[start] = True
    while queue:
        v = queue.popleft()
        print(v, end=' ')
        for i in graph[v]:
            if not visited[i]:
                queue.append(i)
                visited[i] = True


def dfs(graph, v, visited):
    visited[v] = True
    print(v, end=' ')
    for i in graph[v]:
        if not visited[i]:
            dfs(graph, i, visited)

n, m, v = input().split()
n = int(n)
v = int(v)
m = int(m)
visited = [False] * (n+1)

graph = [[] for _ in range(n+1)]
for i in range(m):
    a, b = input().split()
    graph[int(a)].append(int(b))
    graph[int(b)].append(int(a))

for i in range(n+1):
    graph[i].sort()

dfs(graph, v, visited)

print()
visited = [False] * (n+1)
bfs(graph, v, visited)